frappe/desk/form/save.py

```diff
@@ -17,6 +17,10 @@ def savedocs(doc, action):
 	"""save / submit / update doclist"""
 	doc = frappe.get_doc(json.loads(doc))
 	capture_doc(doc, action)
+	if doc.doctype not in ["DocType", "File"] and doc.name.startswith(
+		"new-" + doc.doctype.lower().replace(" ", "-")
```

Comment:
if `doc.__islocal` is set by client we can just assume the name is temporary right? 

This checking of exact naming of temporary names feels weird.

Reply:
yes, just added this as safeguard.